plugins/modules/ds8000_volume_mapping.py

```diff
@@ -0,0 +1,180 @@
+
+import json
+from ansible_collections.ibm.ds8000.plugins.module_utils.ds8000 import (Ds8000ManagerBase, ds8000_argument_spec)
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils._text import to_native
+
+
+class VolumeMapper(Ds8000ManagerBase):
+    def verify_volume_mapping_state(self, volume_id, volume_mapping_state):
+        volume_mapping_on_host = self._get_volume_mapping_on_specific_host()
+        volume_map = None
+        for volume_map in volume_mapping_on_host:
+            if volume_id == volume_map['volume']['id']:
+                volume_mapping_state(volume_id, volume_map)
+                return {'changed': self.changed, 'failed': self.failed}
+        volume_mapping_state(volume_id, volume_map)
+        return {'changed': self.changed, 'failed': self.failed}
+
+    def map_volume_to_host(self, volume_id, volume_map_on_the_host):
+        if volume_map_on_the_host:
+            if volume_id == volume_map_on_the_host['volume']['id']:
+                return
+        name = self.params['name']
+        try:
+            self.client.map_volume_to_host(host_name=name,
+                                           volume_id=volume_id)
+            self.changed = True
+        except Exception as generic_exc:
+            self.failed = True
+            self.module.fail_json(
+                msg="Failed to map volume id {volume_id} to {host_name} host on DS8000 storage. "
+                "ERR: {error}".format(
+                    volume_id=volume_id,
+                    host_name=name,
+                    error=to_native(generic_exc)))
+
+    def _get_volume_mapping_on_specific_host(self):
+        name = self.params['name']
+        volume_mappings = []
+        sub_url = '/hosts/{host_name}/mappings'.format(host_name=name)
+        response = self.get_ds8000_object_from_server(sub_url)
+        parsed_response = json.loads(response.text)['data']['mappings']
+        for volume_map in parsed_response:
+            volume_mappings.append(volume_map)
+        return volume_mappings
+
+    def unmap_volume_from_host(self, volume_id_on_the_host, volume_map):
+        if not volume_map:
+            return
+        if volume_id_on_the_host != volume_map['volume']['id']:
+            return
+        name = self.params['name']
+        lun_id = volume_map['lunid']
+        try:
+            self.client.unmap_volume_from_host(host_name=name,
+                                               lunid=lun_id)
+            self.changed = True
+        except Exception as generic_exc:
+            self.failed = True
+            self.module.fail_json(
+                msg="Failed to unmap volume id {volume_id} from {host_name} host on DS8000 storage."
+                " ERR: {error}".format(
+                    volume_id=volume_map['volume']['id'],
+                    host_name=name,
+                    error=to_native(generic_exc)))
+
+
+def ensure_volume_mapping_state(volume_id, module, volume_mapper):
```

Comment:
The function name and the one above (`verify_volume_mapping_state`) are similar. Thus, I suggest creating two functions for volume_mapper instead of verify_volume_mapping_state:
`ensure_volume_mapped_to_host` and `ensure_volume_unmapped_from_host`
WDYT?

Reply:
done

Reply:
I suggest renaming the "ensure" functions as in the previous comment since they're clearer

Reply:
Also, usually, we name the file and the class accordingly. Why not here?

Reply:
done